unblob/handlers/compression/lzh.py

```diff
@@ -0,0 +1,102 @@
+import io
+from typing import List, Optional
+
+from structlog import get_logger
+
+from ...file_utils import Endian
+from ...models import StructHandler, ValidChunk
+
+logger = get_logger()
+
+PADDING_LEN = 2
+LEVEL_IDENTIFIERS = [0, 1, 2]
+METHOD_IDS = [
+    b"-lh0-",
+    b"-lzs-",
+    b"-lz4-",
+    b"-lh1-",
+    b"-lh2-",
+    b"-lh3-",
+    b"-lh4-",
+    b"-lh5-",
+    b"-lh6-",
+    b"-lh7-",
+    b"-lh8-",
+    b"-lhd-",
+]
+
+
+class LZHHandler(StructHandler):
+
+    NAME = "lzh"
+
+    YARA_RULE = r"""
+        strings:
+            // 2 header bytes followed by method ID
+            $magic = /[\x00-\xff][\x00-\xff](-lh0-|-lzs-|-lz4-|-lh1-|-lh2-|-lh3-|-lh4-|-lh5-|-lh6-|-lh7-|-lh8-|-lhd-)/
```

Comment:
2 comments here for this line.

1.  rename the `$magic` (here and below) to `$lhz_magic` or something. 
This is just to honor [this pull request](https://github.com/IoT-Inspector/unblob/pull/117), I think this make sense to have them in separate names. 

2. I found it a little bit awkward to have the same magic ids defined here as we have stored in the `METHOD_IDS` variable. We can make it work with this change for the YARA rule, so the IDs are filled dynamically:

```python
    YARA_RULE = r"""
        strings:
            // 2 header bytes followed by method ID
            $lhz_magic = /[\x00-\xff][\x00-\xff]({magic_ids})/
        condition:
            $lht_magic
    """.format(magic_ids="|".join([char.decode() for char in METHOD_IDS]))
```
But **I do not want to force this solution**. If you think the duplication is fine for readability (my version is not too reader-friendly), then just leave it!

Reply:
I like how you think. I used your proposition with renamed yara variable 😺